src/components/PostsList/PostsList.tsx

```diff
@@ -1,53 +1,83 @@
-import React from 'react';
-
+import React, { useState, useEffect } from 'react';
 import './PostsList.scss';
+import { getAllPosts, getPosts } from '../../api/posts';
+import { Post } from '../../react-app-env';
 
 type Props = {
-  posts: Post[],
-  postId: number,
-  setPostId: (id: number) => void,
+  userSelectedId: string;
+  selectPost: (arg?: Post) => void;
+  post: Post | undefined;
 };
 
 export const PostsList: React.FC<Props> = ({
-  posts,
-  postId,
-  setPostId,
-}) => (
-  <div className="PostsList">
-    <h2>Posts:</h2>
-
-    {posts.length === 0
-      ? <p>Posts not found</p>
-      : (
-        <ul className="PostsList__list" data-cy="postsList">
-          {posts.map(post => (
-            <li className="PostsList__item" key={post.id}>
+  userSelectedId,
+  selectPost,
+  post,
+}) => {
+  const [currentPostList, setPostList] = useState<Post[]>([]);
+
+  const allPosts = async () => {
+    const result = await getAllPosts();
+
+    setPostList(result);
+  };
+
+  const findposts = async () => {
+    const result = await getPosts(userSelectedId);
+
+    setPostList(result);
+  };
+
+  useEffect(() => {
+    findposts();
+  }, [userSelectedId]);
+
+  useEffect(() => {
+    allPosts();
+  }, []);
```

Comment:
If the purpose of this code is to display all posts when the component is loaded, and then allow the user to filter them by user ID, then this approach may not be correct. This is because every time the userSelectedId changes, the component will reload the list of all posts from the server, which can be redundant and degrade performance.

If you want to load all the posts at the beginning and then let the user filter them, then you don't need to call the allPosts function in the effect. Just call findposts() with the [userSelectedId] dependency, which will run when the userSelectedId changes, and then filter the list of posts according to the selected userSelectedId.
const findPosts = async () => {
  const result = await (userSelectedId ? getPosts(userSelectedId) : getAllPosts());
  setPostList(result);
};

useEffect(() => {
  findPosts();
}, [userSelectedId]);